.github/workflows/changesets.yml

```
@@ -0,0 +1,38 @@
+name: Changesets
+
+on:
+  push:
+    branches: [ main ]
```

It should not run on all commits but be triggered manually 

```suggestion
on: workflow_dispatch
```